### PR TITLE
use Collect() and codeGen.WriteTypes(), now everything works

### PR DESCRIPTION
**Pull request by alexrosenfeld10**
This makes everything work!

### CODE REVIEW

File: AvroSourceGenerator/AvroUtils.cs [hunk 1]
```diff
@@ -8,21 +9,34 @@ namespace AvroSourceGenerator;
...
+            var schema = Schema.Parse(schemaContents, SchemaNames);
+
+            codeGen.AddSchema(schema);
+        }
+
+        codeGen.GenerateCode();
+
+        var tempDir = Path.Join(Path.GetTempPath(), $"avro_generated_{Guid.NewGuid():N}");
+        Directory.CreateDirectory(tempDir);
+        codeGen.WriteTypes(tempDir, true);
```
Comment (alexrosenfeld10) [inline AvroSourceGenerator/AvroUtils.cs:30]: this is super hacky, but it works. I'm not 100% sure it'll work on build machines (GHA, etc) where file permissions might be more restrictive. I'm yet to test that out

File: GeneratedModelsTests/AvroUtilsTest.cs [hunk 2]
```diff
@@ -7,16 +9,14 @@ public class AvroUtilsTest
...
-        var solarSystemSchema = File.ReadAllText("./solar-system-model.json");
-        var solarSystemSourceCode = AvroUtils.GenerateSourceCode(solarSystemSchema);
+        var sourceCode = AvroUtils.GenerateSourceCode(schemaNamesAndContents);
 
-        // we should define the planet enum in one file
-        Assert.Contains("public enum PlanetEnum", planetEnumSourceCode);
-        
-        // we should *reference*, but not define, the planet enum in the other file
-        Assert.DoesNotContain("public enum PlanetEnum", solarSystemSourceCode);
+        Assert.True(Regex.Matches(sourceCode, "public enum PlanetEnum").Count == 1);
```
Comment (alexrosenfeld10) [inline GeneratedModelsTests/AvroUtilsTest.cs:20]: we only get one!

File: GeneratedModelsTests/ModelsTest.cs [hunk 3]
```diff
@@ -1,18 +1,51 @@
...
+            HomePlanet = PlanetEnum.Earth
+        };
+        Assert.Equal(PlanetEnum.Earth, rocketMcRocketFace.HomePlanet);
+    }
+
+    [Fact]
+    public void SolarSystemModelExists()
+    {
+        // this test won't run unless the source generation worked
+        var ourSolarSystem = new SolarSystem
```
Comment (alexrosenfeld10) [inline GeneratedModelsTests/ModelsTest.cs:34]: these models exist now!

File: AvroSourceGenerator/AvroGenerator.cs [hunk 4]
```diff
@@ -33,13 +33,13 @@ public void Initialize(IncrementalGeneratorInitializationContext initContext)
...
+                    (name: Path.GetFileNameWithoutExtension(file.Path),
+                        content: file.GetText(cancellationToken)!.ToString()))
+            .Collect();
 
         initContext.RegisterSourceOutput(schemaFileNameAndContents,
-            static (spc, nameAndContent) =>
-                spc.AddSource($"AvroGeneratedSchemas.{nameAndContent.name}",
-                    AvroUtils.GenerateSourceCode(nameAndContent.content)));
+            static (spc, namesAndContents) =>
+                spc.AddSource("AvroGeneratedSchemas", AvroUtils.GenerateSourceCode(namesAndContents)));
```
Comment (alexrosenfeld10) [inline AvroSourceGenerator/AvroGenerator.cs]: I don't _love_ glomming everything into a single file, feels like the kind of thing where a single file size limit might get hit one day. Needs more investigation. I can also just change some of the logic here to separate things out into separate files

**Comment by tcardwell-seismic** [comment]
Technically matrix owns all of our SEB schemas, right? Should this code just be part of matrix rather than a new repo? (Unless I'm misunderstand the purpose of this code)

**Comment by alexrosenfeld10** [comment]
@tcardwell-seismic this is a public repo I'm using to POC a fork of avro that supports generating c# code for cross-file schema definitions. The idea being, we can DRY common types / enums / building blocks up into common namespaces, and use them uniformly across various top level schemas.

Ultimately, when the fork is merged to upstream `apache/avro`, we'll pull this into our tooling.

**Pull request merged**